core/fetcher/fetcher.go

```diff
@@ -130,14 +138,59 @@ func (f *Fetcher) fetchAttesterData(ctx context.Context, slot int64, defSet core
 	return resp, nil
 }
 
+func (f *Fetcher) fetchAggregatorData(ctx context.Context, slot int64, defSet core.DutyDefinitionSet) (core.UnsignedDataSet, error) {
+	resp := make(core.UnsignedDataSet)
+	for pubkey := range defSet {
+		// Query AggSigDB for DutyPrepareAggregator to get beacon committee subscription.
+		prepAggData, err := f.aggSigDBFunc(ctx, core.NewPrepareAggregatorDuty(slot), pubkey)
+		if err != nil {
+			return core.UnsignedDataSet{}, err
+		}
+
+		sub, ok := prepAggData.(core.SignedBeaconCommitteeSubscription)
+		if !ok {
+			return core.UnsignedDataSet{}, errors.New("invalid beacon committee subscription")
+		}
+
+		res, err := eth2exp.CalculateCommitteeSubscriptionResponse(ctx, f.eth2Cl, &sub.BeaconCommitteeSubscription)
+		if err != nil {
+			return core.UnsignedDataSet{}, err
+		}
+
+		if !res.IsAggregator {
+			continue
+		}
+
+		// Query AggSigDB for DutyAttester to get attestation data root.
+		attData, err := f.aggSigDBFunc(ctx, core.NewAttesterDuty(slot), pubkey)
+		if err != nil {
+			return core.UnsignedDataSet{}, err
+		}
+
+		att, ok := attData.(core.Attestation)
+		if !ok {
+			return core.UnsignedDataSet{}, errors.New("invalid attestation")
+		}
+
+		// Query BN for aggregate attestation.
+		aggAtt, err := f.eth2Cl.AggregateAttestation(ctx, eth2p0.Slot(slot), att.Data.BeaconBlockRoot)
+		if err != nil {
+			return core.UnsignedDataSet{}, err
+		}
+
+		resp[pubkey] = core.AggregatedAttestation{
```

Comment:
Isn't there a `core.NewAggregateAttesation(aggAtt)` function?

Reply:
No. I created a new type for aggregate attestation implementing unsignedData
```go
// AggregatedAttestation wraps Attestation and implements the UnsignedData interface.
type AggregatedAttestation struct {
	eth2p0.Attestation
}
```